### DISCUSSION TITLE
Stops training after one epoch

### DISCUSSION THREAD

**Discussion by BramVanroy** [Lightning-AI/pytorch-lightning#8452] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi all

I am new to PL so this might be an obvious question. Could not find an answer in the extensive docs. I am training a classification problem (CrossEntropyLoss) and this is my trainer init.

```python
early_stopping = EarlyStopping("valid_loss", min_delta=0.1, mode="min")
trainer = pl.Trainer(gpus=1, callbacks=[early_stopping], flush_logs_every_n_steps=50)
```

I had expected the script to run indefinitely until early stopping kicks in. But after the first training epoch finishes (around 250 steps) and validation is done, the script just stops. Is that to be expected? It does not give any feedback or error message but the script just stops running as if it was "done executing". (It does not freeze, it is just "done".)

How can I make the script run indefinitely until the Early Stopping criterion is met?

**Comment by tchaton** [comment]
Dear @BramVanroy,

Yes, the script should keep training. Might providing a reproducible script using the BoringModel: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py.

Let's convert this discussion into an issue when you have a reproducible script :)

Best,
T.C
